Add multi-agent training and evaluation example using PPO with Ray

vteran93 opened the pull request.

SergioSV96 committed:
Add multi-agent training and evaluation example using PPO with Ray

vteran93 commented on examples/multiagent_simplified.py:
```diff
@@ -0,0 +1,316 @@
+
+    def __init__(self, env_config=None):
+        config = env_config or {}
+        self.env = F110Env(config=config, render_mode=config.get("render_mode"))
```
No es que me quiera poner a juzgar las formas, pero porque lo hiciste con instancias de F110Env en vez de heredar?

SergioSV96 replied:
Buena pregunta realmente, el problema es que si haces herencia, habría que hacer herencia múltiple, tanto de F110Env como del MultiAgentEnv, lo cual entran en conlficto, de esta forma la clase actúa como un proxy, separamos los agentes y se los devolvemos a la clase MultiAgentEnv, por debajo llamando al f110env.


`class SimpleMultiAgentF110(MultiAgentEnv):`

SergioSV96 committed:
Update dependencies in pyproject.toml to latest versions

SergioSV96 committed:
Add tensorboard script

SergioSV96 committed:
Reduce save frequency in training loop from 2000 to 200 timesteps, this makes it so tensorboard logs each 200 timesteps

vteran93 committed:
Implement sac with ray like Sergio

SergioSV96 committed:
Merge pull request #4 from vteran93/sac_ray_victor

Implement sac with ray like Sergio

SergioSV96 committed:
Enhance reward calculation in MultiAgentF110 with progress-based rewards and collision penalties

SergioSV96 committed:
Better organize the repo, renaming files and some refactoring

SergioSV96 committed:
Add reward calculation logic and TensorBoard integration for multi-agent training

- Implemented a reward calculation method in `reward.py` to evaluate agent performance based on progress, collision penalties, and survival rewards.
- Created a new Jupyter notebook `tensorboard_test.ipynb` to facilitate training and evaluation of the multi-agent PPO setup, including logging and visualization of training metrics.

SergioSV96 committed:
Implement code changes to enhance functionality and improve performance
